Complete move to net70 ref assemblies

jaredpar opened the pull request.

jaredpar commented:
@dotnet/roslyn-compiler, @jcouv PTAL

jaredpar commented on src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs:
```diff
@@ -12054,11 +12055,13 @@ void I1.M1()
 
                 compilation3.VerifyDiagnostics();
 
+#if NETCOREAPP
```
@AlekseyTs wanted to make sure you reviewed both uses of `#if NETCOREAPP` in this change. These `CompileAndVerify` calls are executing DIM methods and that should only be legal on .NET Core. Yet this was running on .NET Framework as well. I'm not quite sure how it was running before my change as that shouldn't work.

This change _feels_ correct but I'm also hesitant as it was running before so I wanted to make sure you looked this part over.

jaredpar commented on src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs:
```diff
@@ -50163,9 +50168,6 @@ public sealed class TypeIdentifierAttribute : Attribute
         [Fact]
         public void NoPia_01()
         {
-            var attributes = CreateCompilation(NoPiaAttributes, options: TestOptions.ReleaseDll, targetFramework: TargetFramework.Net50);
-            var attributesRef = attributes.EmitToImageReference();
```
This pattern was removed in the file because `TargetFramework.Net50` now includes all the standard references which brings in `System.Runtime.InteropServices.dll` which has this definition.

jaredpar commented on src/Compilers/Test/Core/TargetFrameworkUtil.cs:
```diff
@@ -176,16 +149,24 @@ public static class TargetFrameworkUtil
+
+        static TargetFrameworkUtil()
+        {
+            Debug.Assert(GetReferences(TargetFramework.NetCoreApp).SequenceEqual(NetCoreApp.References));
```
This will ensure going forward that we keep both definitions of `NetCoreApp` in sync.

AlekseyTs commented on src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs:
```diff
@@ -6,6 +6,7 @@
 
 using System;
 using System.Linq;
+using Basic.Reference.Assemblies;
```
Is this using necessary?

jaredpar replied:
Yes, the use of `Net70.SystemRuntime` requires it

tmat commented on src/Compilers/Test/Core/TargetFrameworkUtil.cs:
```diff
@@ -7,11 +7,13 @@
 using static TestReferences;
 using static Roslyn.Test.Utilities.TestMetadata;
 using Basic.Reference.Assemblies;
+using System.Diagnostics;
```
>Diagnostics [](http://example.com/codeflow?start=13&length=11)

sort usings

tmat reviewed (approved):
:shipit:

jaredpar commented:
@AlekseyTs addressed the feedback in the `DefaultInterfaceImplementationTests`. The test case `NoPia_09` is more complex than necessary because I hit a retargeting issue (#65048) that forced a precise handling of references.

AlekseyTs commented:
> addressed the feedback in the `DefaultInterfaceImplementationTests`.

Changes in  `DefaultInterfaceImplementationTests` LGTM.

jaredpar commented:
@dotnet/roslyn-compiler PTAL. Have some other changes stacked up behind this

cston commented on src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs:
```diff
@@ -48,15 +48,26 @@ private static CSharpCompilation CreateCompilationWithFunctionPointers(CSharpTes
+            var references = includeUnmanagedCallersOnly
+                ? TargetFrameworkUtil.GetReferences(TargetFramework.Net50)
+                : TargetFrameworkUtil.GetReferencesWithout(TargetFramework.Net50, "System.Runtime.InteropServices.dll");
+            return CreateCompilation(source, references: references, options: options ?? TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.Regular9, targetFramework: TargetFramework.Empty);
```
>parseOptions: TestOptions.Regular9 [](http://example.com/codeflow?start=119&length=34)

Not specific to this PR, but are we testing function pointers with `-langversion:latest`, or are most tests using `-langversion:9`?
